lxd/storage_utils.go

```diff
@@ -181,13 +181,25 @@ func lxdUsesPool(dbObj *sql.DB, onDiskPoolName string, driver string, onDiskProp
 	return false, "", nil
 }
 
+func xfsGenerateNewUUID(path string) error {
+	_, err := shared.RunCommand("xfs_admin", "-U", "generate", path)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
 func makeFSType(path string, fsType string) (string, error) {
 	var err error
 	var msg string
 
 	switch fsType {
 	case "xfs":
 		msg, err = shared.TryRunCommand("mkfs.xfs", path)
+		if err == nil {
+			err = xfsGenerateNewUUID(path)
+		}
```

Comment:
My understanding is that this is the one time where this isn't actually needed.

XFS requires all mounted filesystems to have differing UUIDs. So the time where you really need to generate a new UUID is after cloning an existing filesystem. Creating a new one through mkfs.xfs should get you a clean UUID and so not require this step.

Reply:
That point makes sense. I should not send patches at 6 in the morning.